### DISCUSSION TITLE
Standardization needs across implementation teams

### DISCUSSION THREAD

**Discussion by kaitlin-beegle** [filecoin-project/FIPs#217] [TPM Best Practices]
At [Core Devs Meeting #31](https://github.com/filecoin-project/tpm/blob/master/Core%20Dev%20Meetings/Meeting%200031.md), the Venus and Lotus teams noted the need for improved standardization of certain non-protocol features, tools, and interfaces.  For example: 

- API endpoints
- RPC interfaces 
- Lotus' query ask protocol 

Outstanding questions include: 

1. What other items, etc., would benefit from standardization across implementation teams?  
2. How should we define standardization needs in the documentation? 
3. What RPC/API models or best practices do other Web3 projects use? 

These are big, general questions, but the intent was to carve out space where we can start a conversation!

**Comment by arajasek** [comment]
The folks working on Boost might have ideas for other things worth standardizing.

Lotus is planning on stabilizing its V1 API in Q1 2022. Since most other impls already match Lotus's API fairly closely, that could be a good point to launch this effort. Subsequent changes could be discussed and adopted by the core devs, or simply marked as deviations from the standard if any one impl wanted to do so.
